plugins/module_utils/foreman_helper.py

```diff
@@ -371,6 +387,21 @@ def _update_entity(self, resource, desired_entity, current_entity, params, entit
         else:
             return False, current_entity
 
+    def _revert_entity(self, resource, current_entity, params):
```

Comment:
Do we want this in the `foreman_helper`? I think we can catch this case in the module before calling `ensure_entity`. Open for discussion.

Reply:
@evgeni suggested it to be in ensure_resource_state. But if it is specific to snapshot only then it can be done in module.

Reply:
If he said that... I am fine with either ways.

Reply:
I remember asking if it *can* be inside the regular ensure path. and looking at the code it cleanly fits into it, so I'm fine with it being here.

I wonder if we should guard that part by `if resource == 'snapshots'` and fail if it's not, but I don't think it's strictly necessary.